src/main/java/com/epam/javacourses2016/task10/SolverTask10.java

```diff
@@ -13,10 +17,35 @@
 
     /**
      * Подсчитывает количество вхождений каждого слова в файле.
+     *
      * @param input Файл с исходными данными.
      * @return Множество пар <слово, количество вхождений в файле>.
      */
     public HashMap<String, Integer> countNumberWords(File input) {
-        return null;
+        HashMap<String, Integer> uniqueWords = new HashMap<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(input))) {
+            String line;
+            int y;
+            Pattern pattern = Pattern.compile("[(A-Za-z)('\\-)(A-Za-z)]+");
+            while ((line = reader.readLine()) != null) {
+                Matcher matcher = pattern.matcher(line);
+                while (matcher.find()) {
```

Comment:
Yes, it will be better. Didn't notice Scanner#Next could use regex pattern for strings.
As for regex pattern, i thought we should remove numbers from text if there is some(1,2,3 etc).